### PR TITLE
Add basic field remapping support

### PR DESCRIPTION
**Pull request by Juuxel**

### CODE REVIEW

File: unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java [hunk 1]
```diff
@@ -70,12 +90,50 @@ public void visitLineNumber(int lineNumber)
...
 	public void visitSimpleConstantDefinition(String group, String owner, String name, String value, String descriptor)
 	{
-		delegate.visitSimpleConstantDefinition(group, owner, name, value, descriptor);
+		//Reassigning the parameters tends to cause bugs
+		String remappedOwner = owner,
+			   remappedName = fieldMappings.getOrDefault(new FieldKey(owner, name), name),
+			   remappedDescriptor = descriptor;
+
+		if (classMappings.containsKey(owner))
+			remappedOwner = classMappings.get(owner);
```
Comment (liach) [inline unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java:99]: Can't you just do a `classMappings.getOrDefault(owner, owner)` when you declare `remappedOwner` local var?

File: unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java [hunk 2]
```diff
@@ -70,12 +90,50 @@ public void visitLineNumber(int lineNumber)
...
 	public void visitFlagConstantDefinition(String group, String owner, String name, String value, String descriptor)
 	{
-		delegate.visitFlagConstantDefinition(group, owner, name, value, descriptor);
+		//Reassigning the parameters tends to cause bugs
+		String remappedOwner = owner,
+			   remappedName = fieldMappings.getOrDefault(new FieldKey(owner, name), name),
+			   remappedDescriptor = descriptor;
+
+		if (classMappings.containsKey(owner))
+			remappedOwner = classMappings.get(owner);
```
Comment (liach) [inline unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java:123]: Same comment as above

**Comment by modmuss50** [comment]
Quite a lot of duplicated logic here, should be easy enough to split some of it out into a method? i.e the descriptor remapping?

File: unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java (hunk 1 above)
Comment (Juuxel) [reply unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java]: This was actually copied straight from the target remapping method, but yeah I switched everything to `getOrDefault`.

**Comment by Juuxel** [comment]
@modmuss50 I split everything out to their own methods, should be much cleaner now.

**Comment by Earthcomputer** [comment]
This contains a breaking change to unpick's API, is that okay? This is why I was hesitant to do this myself when looking into fields.

**Comment by Juuxel** [comment]
It's not a breaking change. The old constructor for `UnpickV2Remapper` is retained, with `fieldMappings` just defaulting to an empty map.

File: unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java [hunk 3]
```diff
@@ -31,30 +34,66 @@ public UnpickV2Remapper(Map<String, String> classMappings, Map<MethodKey, String
 	{
 		this.classMappings = classMappings;
 		this.methodMappings = methodMappings;
+		this.fieldMappings = Collections.emptyMap();
 		this.delegate = delegate;
```
Comment (modmuss50) [inline unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java:38]: [NIT] Should really call the other constructor here.

**Review by modmuss50 — APPROVED** [review]
Looks good 👍  Thanks, I will test this locally before merging.

File: unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java (hunk 3 above)
Comment (Juuxel) [reply unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java:38]: Yeah I also realised that after writing this, I can change it if you want to.

Comment (modmuss50) [reply unpick-format-utils/src/main/java/daomephsta/unpick/constantmappers/datadriven/parser/v2/UnpickV2Remapper.java:38]: Sure, easy enough to do

**Review by liach — APPROVED** [review]

**Pull request merged**